src/utils/promise.js

```diff
@@ -0,0 +1,13 @@
+const resolveMappedPromises = async (promises) => {
+  const keys = Object.keys(promises);
+  const acc = {};
+  for (let i = 0; i < keys.length; i += 1) {
+    const result = await promises[keys[i]];
+    acc[keys[i]] = result;
```

Comment:
why not just Promise.all(promises) ?

If I read this correctly, this... waits for each promise, sequentially?

Reply:
The promises get fired in parent function, this basically just assigns correct values to object keys, much like Promise.all